vm/src/stdlib/signal.rs

```diff
@@ -81,50 +94,68 @@ fn alarm(time: PyIntRef, _vm: &VirtualMachine) -> u32 {
     prev_time.unwrap_or(0)
 }
 
-#[allow(clippy::needless_range_loop)]
-pub fn check_signals(vm: &VirtualMachine) {
-    for signum in 1..NSIG {
-        let triggerd = unsafe { TRIGGERS[signum].swap(false, Ordering::Relaxed) };
+pub fn check_signals(vm: &VirtualMachine) -> PyResult<()> {
+    for (signum, trigger) in TRIGGERS.iter().enumerate().skip(1) {
+        let triggerd = trigger.swap(false, Ordering::Relaxed);
         if triggerd {
-            let handler = vm
-                .signal_handlers
-                .borrow()
-                .get(&(signum as i32))
-                .expect("Handler should be set")
-                .clone();
-            vm.invoke(&handler, vec![vm.new_int(signum), vm.get_none()])
-                .expect("Test");
+            let handler = &vm.signal_handlers.borrow()[signum];
+            if vm.is_callable(handler) {
```

Comment:
How can the handler be not callable? We check this when we set the signal.

Reply:
If it's `SIG_DFL` or `SIG_IGN`, that still gets stored in the array and returned as the old handler when you call `signal()`.